src/HealthGPS.Console/jsonparser.h

```diff
@@ -103,4 +115,24 @@ template <typename T> void from_json(const nlohmann::json &j, std::optional<T> &
     }
 }
 
+template <typename T>
+void to_json(nlohmann::json &j, const std::optional<hgps::core::Interval<T>> &p) {
+    if (p) {
+        j = *p;
+    } else {
+        // Null interval expressed as empty JSON array
+        j = nlohmann::json::array();
+    }
+}
+
+template <typename T>
+void from_json(const nlohmann::json &j, std::optional<hgps::core::Interval<T>> &p) {
+    // Treat null JSON values and empty arrays as a null Interval
+    if (j.is_null() || (j.is_array() && j.empty())) {
+        p = std::nullopt;
+    } else {
+        p = j.get<hgps::core::Interval<T>>();
+    }
+}
+
```

Comment:
That's true, but currently the code uses empty arrays to indicate a null interval rather than a JSON null value, which is really what it should be, so I added these helpers to avoid introducing a breaking change. I meant to add a comment saying as much, but forgot.

That said, I don't like this and I think the right solution is to do away with this special-casing and just use null values instead, though that would mean changing config files too. Shall I open an issue for it?

Reply:
Agree that the null values are preferred. I think I already started using null values elsewhere for things. /Go ahead.